src/ViewModel/MessageGroup.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace eLife\Patterns\ViewModel;
+
+use eLife\Patterns\ArrayAccessFromProperties;
+use eLife\Patterns\ArrayFromProperties;
+use eLife\Patterns\CastsToArray;
+use InvalidArgumentException;
+
+final class MessageGroup implements CastsToArray
+{
+    use ArrayAccessFromProperties;
+    use ArrayFromProperties;
+
+    private $id;
+    private $errorText;
+    private $infoText;
+
+    public function __construct(
+        string $errorText = null,
+        string $infoText = null
+    ) {
+        if (empty($errorText) && empty($infoText)) {
+            throw new InvalidArgumentException('A MessageGroup must contain at least one message.');
```

Comment:
We might want to consider have two constructors instead: one for info text + optional error text, and one for error text only.